lib/utils.js

```diff
@@ -141,16 +145,34 @@ exports.getXpiInfoFromManifest = getXpiInfoFromManifest;
 function getXpiInfo(xpiPath) {
   return extractXPI(xpiPath)
     .then(function(tmpXPI) {
-      var manifest = {};
-      try {
-        manifest = require(path.join(tmpXPI.path, "package.json"));
-      } catch (e) {}
-
-      tmpXPI.remove();
+      var getInfo;
+      var packageFile = path.join(tmpXPI.path, "package.json");
+      var installRdfFile = path.join(tmpXPI.path, "install.rdf");
 
-      // TODO: when there's no manifest, parse install.rdf to get info.
+      if (fileExists(packageFile)) {
```

Comment:
If it isn't a jpm it might have a package.json just to install jpm with `npm install`, for example. But there could be dependencies that get bundled as well.